amp.php

```diff
@@ -307,6 +307,9 @@ function amp_deactivate() {
 	flush_rewrite_rules( false );
 }
 
+// The plugins_loaded action is the earliest we can run this since that is when pluggable.php has been required and wp_hash() is available.
```

Comment:
Why not replace `wp_hash` with something else, then? It doesn't need to be cryptographically secure, so any type of hash will work, really.

Reply:
I think it would even make sense to include a hash function in the `amp/common` library, so that we have consistent hashing across all libraries.

Reply:
Good question. It's not just purely a hash that we need. It's hashing using one of the WordPress secrets, which entails using `wp_salt()` which is in turn another pluggable function. Then the input for hashing is `wp_nonce_tick()` which is yet another pluggable function. So these three functions all require this to be run at `plugins_loaded` and I believe we need to depend on them specifically.

https://github.com/ampproject/amp-wp/blob/1c3c8539aeb59b564da3bc6db16584dfc25dd88f/includes/validation/class-amp-validation-manager.php#L1773-L1775